From c34dca74854cb6e7f2b09affa2a4ab0145e62371 Mon Sep 17 00:00:00 2001
From: Kartik Mahajan <stonebrakert6@gmail.com>
Date: Fri, 30 Jun 2023 11:34:07 +0530
Subject: [PATCH] Fixes build failure on ./configure && make
 --shuffle=2836571325

./configure && make --shuffle=2836571325 fails.
The build fails on parallel builds for me and above command reproduces it.
Note that --shuffle options is available on make version >= 4.4
Target ffi_libname in src/Makefile should have a additional dependency
upon $(liburing_sobjs)

Upstream-Status: Backport [https://github.com/axboe/liburing/commit/c34dca74854cb6e7f2b09affa2a4ab0145e62371]
Signed-off-by: Kartik Mahajan <stonebrakert6@gmail.com>
---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index c4c28cb..7832095 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -81,7 +81,7 @@ liburing-ffi.a: $(liburing_objs) $(liburing_ffi_objs)
 $(libname): $(liburing_sobjs) liburing.map
 	$(QUIET_CC)$(CC) $(SO_CFLAGS) -shared -Wl,--version-script=liburing.map -Wl,-soname=$(soname) -o $@ $(liburing_sobjs) $(LINK_FLAGS)
 
-$(ffi_libname): $(liburing_ffi_objs) $(liburing_ffi_sobjs) liburing-ffi.map
+$(ffi_libname): $(liburing_ffi_objs) $(liburing_ffi_sobjs) $(liburing_sobjs) liburing-ffi.map
 	$(QUIET_CC)$(CC) $(SO_CFLAGS) -shared -Wl,--version-script=liburing-ffi.map -Wl,-soname=$(ffi_soname) -o $@ $(liburing_sobjs) $(liburing_ffi_sobjs) $(LINK_FLAGS)
 
 install: $(all_targets)
-- 
2.41.0

